eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -53,15 +57,31 @@ case class AuditResponse(sent: Seq[PaymentSent], received: Seq[PaymentReceived],
 
 case class TimestampQueryFilters(from: Long, to: Long)
 
+case class MutualCloseStatus(unpublished: Satoshi, unconfirmed: Satoshi, confirmed: Satoshi)
```

Comment:
Actually, this is just additional information about mutual close transactions, useful to track evictions. The mutual close balance is tracked as part of onchain balance, see:

```scala
  /**
   * Mutual close transactions are always immediately published, and will be taken into account by bitcoin core, so we
   * don't count them in the balance.
   */
  case class ClosingBalance
```

But now that I think of it, we should probably treat mutual closes like other transactions, in the offchain balance, and prune it afterwards. The goal of pruning was initially to remove duplicates for transactions that were published (and possibly confirmed but less than mindepth), but it should also nicely handle evicted transactions.

Reply:
Ok, I remember now why the mutual close transactions were handled like this. The problem is that there can be several competing mutual closes, and in that case the simple pruning logic that we have to make sure we count the amount exactly once doesn't work. This would lead us to possibly count twice the amount in the competing-mutual-closes scenario.

On the other hand, what we had before meant that we simply didn't count the amount in the competing-mutual-closes scenario, and also in the case where the mutual close was evicted. Not perfect either.

In general I would say that it is better to undershoot rather than overshoot the balance computation, but here the trade-off makes sense for consistency, and also it is easy to tell when we are potentially counting a mutual close twice: the `mutualCloseBalance` should always be zero after pruning.

So I went ahead and did it with 4b7839203, with a detailed comment on what is going on.